modules/post/windows/gather/credentials/foxmail.rb

```diff
@@ -53,18 +53,18 @@ def find_string(file, offset, length = 0)
   def enum_session_file(fpath)
     account_paths = []
     session.fs.dir.foreach(fpath) do |mail_addr|
-      account_path = fpath + '\\' + mail_addr + '\\Accounts\\Account'
-      account_paths.push(account_path + '.rec0') if session.fs.file.exist?(account_path + '.rec0')
-      account_paths.push(account_path + '.stg') if session.fs.file.exist?(account_path + '.stg')
-      account_paths.push(account_path + '.tdat') if session.fs.file.exist?(account_path + '.tdat')
+      account_path = "#{fpath}\\#{mail_addr}\\Accounts\\Account"
+      account_paths.push("#{account_path}.rec0") if session.fs.file.exist?("#{account_path}.rec0")
+      account_paths.push("#{account_path}.stg") if session.fs.file.exist?("#{account_path}.stg")
+      account_paths.push("#{account_path}.tdat") if session.fs.file.exist?("#{account_path}.tdat")
     end
     tbl = []
     print_status("Search account files on #{fpath}")
 
     # enum session file
     account_paths.each do |file_name|
       file = read_file(file_name) if session.fs.file.exist?(file_name)
-      if file.nil? || file.empty?
+      if file.nil? || file.empty? || file[0, 4] != 'RECF'
```

Comment:
All of the changes in this file look good but I was a bit curious as to what this fix was for, do you mind explaining?

Reply:
This flag ensures that this is a real configuration file, but I can’t find the old version of the installer, maybe this is suitable for the current new version, if it is compatible with other versions, I think this code should be deleted

Reply:
@cn-kali-team Sorry are you saying you think this should be deleted? I do see this will check and help ensure that the file is correct by ensuring the starting characters match what is expected, which will work on the current version of FoxMail. I will see if this works on older versions.

Also you can find older versions at hxxps://foxmail.en.uptodown.com/windows/versions if you want to test your module against older versions.

Reply:
Ok so I tested this against a few other versions. Looks like version 6.5 build 023 uses a different file format and saves it into a file under `C:\Program Files (x86)\Foxmail\mail\foocarddog@gmail.com\Account.stg`.

Also some of the older version 7 ones, when I asked it to install to the `C:\FoxMail` directory, ended up saving the account info in `C:\Foxmail\Data\Accounts\Account.rec0`. So this may be something to keep in mind.

That being said all the version 7 installations appear to have a .rec0 file that starts with the RECF signature you mentioned so I think it should be okay to keep that in there if we also update the module documentation to note that this module only works on version 7.x of FoxMail.

Reply:
So should I add support for older versions? I don't think anyone will use the old version of Foxmail

Reply:
@cn-kali-team Its fine to only have support for the 7.x versions but you should note this in your module description. Otherwise people may falsely believe that this module works on all versions of FoxMail including versions such as 6.5, which will not be the case.